frontend/src/test/java/com/google/sps/InMemoryPlaceVisitStorageTest.java

```diff
@@ -0,0 +1,348 @@
+   * userMark of "must-see" or "if-time" are returned in list when there is only
+   * one unique tripId in storage
+   */
+  @Test
+  public void listTripPlaceVIsits_oneTripInStorage_returnsOnlyTripsWithCorrectMark()
+      throws PlaceVisitAlreadyExistsException {
+    InMemoryPlaceVisitStorage storage = new InMemoryPlaceVisitStorage();
+    storage.addPlaceVisit(LONDON);
+    storage.addPlaceVisit(PARIS);
+    storage.addPlaceVisit(ROME);
+    storage.addPlaceVisit(TOKYO);
+    List<PlaceVisitModel> tripAPlaces = storage.getTripPlaceVisits("a");
+    assertThat(tripAPlaces).containsExactly(ROME, TOKYO, PARIS);
+  }
+
+  /**
+   * test that only the PlaceVisitModel objects with the given tripId and
+   * userMark of "must-see" or "if-time" are returned in list when there are
+   * multiple unique tripIds in storage
+   */
+  @Test
+  public void listTripPlaceVisits_multipleTripsInStorage_returnsOnlyTripsWithCorrectTripIdAndMark()
+      throws PlaceVisitAlreadyExistsException {
+    InMemoryPlaceVisitStorage storage = new InMemoryPlaceVisitStorage();
+    storage.addPlaceVisit(LONDON);
+    storage.addPlaceVisit(PARIS);
+    storage.addPlaceVisit(ROME);
+    storage.addPlaceVisit(TOKYO);
+    storage.addPlaceVisit(BEIJING);
+    storage.addPlaceVisit(TOKYO_B);
+    storage.addPlaceVisit(SEOUL);
+    List<PlaceVisitModel> tripAPlaces = storage.getTripPlaceVisits("a");
+    assertThat(tripAPlaces).containsExactly(ROME, TOKYO, PARIS);
+  }
+
+  /**
+   * test that no PlaceVisitModel objects are returned in list when there are no
+   * PlaceVisitModel objects in storage with the given tripId
+   */
+  @Test
+  public void listTripPlaceVisits_tripNotInStorage_returnsEmptyList()
+      throws PlaceVisitAlreadyExistsException {
+    InMemoryPlaceVisitStorage storage = new InMemoryPlaceVisitStorage();
+    storage.addPlaceVisit(LONDON);
+    storage.addPlaceVisit(PARIS);
+    storage.addPlaceVisit(ROME);
+    storage.addPlaceVisit(TOKYO);
+    storage.addPlaceVisit(BEIJING);
+    storage.addPlaceVisit(TOKYO_B);
+    storage.addPlaceVisit(SEOUL);
+    List<PlaceVisitModel> tripAPlaces = storage.getTripPlaceVisits("c");
+    assertThat(tripAPlaces).isEmpty();
+  }
+
+  /**
+   * test that no PlaceVisitModel objects are returned in list after removing
+   * PlaceVisitModel objects with the given tripId
+   */
+  @Test
+  public void removeTripPlaceVisits_getTripPlaceVisits_tripInStorage_returnsEmptyList()
+      throws PlaceVisitAlreadyExistsException, TripNotFoundException {
+    InMemoryPlaceVisitStorage storage = new InMemoryPlaceVisitStorage();
+    storage.addPlaceVisit(LONDON);
+    storage.addPlaceVisit(PARIS);
+    storage.addPlaceVisit(ROME);
+    storage.addPlaceVisit(TOKYO);
+    storage.addPlaceVisit(BEIJING);
+    storage.addPlaceVisit(TOKYO_B);
+    storage.addPlaceVisit(SEOUL);
+    storage.removeTripPlaceVisits("a");
+    List<PlaceVisitModel> tripAPlaces = storage.getTripPlaceVisits("a");
```

Comment:
In order to make tests more precise (i.e., test only the functionality you're trying to test in each case) and easier to refactor, it's best practice to only insert the place visit that you're deleting and querying for. In particular, you do not need to insert visits from trip `b` to test this. The only reason to do so is if you think that the two should interact with each other (or if you're concerned that there's a risk that place visits from unrelated trips might be returned in this case).

(This general principle applies to all test cases.)